include/zephyr/kernel_structs.h

```diff
@@ -60,9 +60,15 @@ extern "C" {
 /* Thread is suspended */
 #define _THREAD_SUSPENDED (BIT(4))
 
-/* Thread is being aborted */
+/* Thread is in the process of aborting */
 #define _THREAD_ABORTING (BIT(5))
 
+/* Thread is in the process of suspending */
+#define _THREAD_SUSPENDING (BIT(6))
+
+/* Thread is in the process of halting */
+#define _THREAD_HALTING    (_THREAD_ABORTING | _THREAD_SUSPENDING)
```

Comment:
Mentioned this long ago, but still don't love this API scheme "_THREAD_HALTING" looks like all the other enumerants, but it's not, it's a multi-bit mask and you can't use it like that.  Would prefer to see this "|" operator down inside "is_halting()" and e.g. "set_halting()", which is how other thread state predicates work.

Reply:
I missed doing that before. Will correct.